addons/stock/models/stock_picking.py

```diff
@@ -712,6 +712,12 @@ def create(self, vals):
             if picking_type.sequence_id:
                 vals['name'] = picking_type.sequence_id.next_by_id()
 
+        # make sure to write the picking partner_id into the SM
+        if vals.get('partner_id'):
+            for move in vals.get('move_lines', []) + vals.get('move_ids_without_package', []):
```

Comment:
On 15.0, the this for-loop does no more exist (since 515b9a78e66049f12a09b9150c0e77ccb925de02). The locations of the picking's SMs are defined by the UI:
https://github.com/odoo/odoo/blob/f93c35b7021144a0fca26fb80bd4c696d60e182b/addons/stock/views/stock_picking_views.xml#L358-L366
We should follow the same idea (as you did as first fix version 7ee84dc100474408d0d4299356e0492ede582eb2)